### PR TITLE
[DM] [BTAT-8064] Added email verified check for contact pref journey

### PR DESCRIPTION
**Pull request by TheDeadlyPianist**

### CODE REVIEW

File: conf/app.routes [hunk 1]
```diff
@@ -1,92 +1,96 @@
...
 
 # contact preference confirm email
-GET         /preference-confirm-email                   controllers.contactPreference.EmailToUseController.show
-POST        /preference-confirm-email                   controllers.contactPreference.EmailToUseController.submit
+GET         /preference-confirm-email              controllers.contactPreference.EmailToUseController.show
+POST        /preference-confirm-email              controllers.contactPreference.EmailToUseController.submit
+
+# check if email verification has been sent
+GET         /contact-preference/update-email-address        controllers.contactPreference.EmailVerificationController.checkVerificationStatus
+GET         /contact-preference/update-contact-pref          controllers.contactPreference.EmailVerificationController.updateContactPrefEmail
```
[the remark this answers is not in the document]
Comment (TheDeadlyPianist) [reply conf/app.routes:83]: I'll take a look. My apologies. Didn't pay much attention when rebasing this file by the looks of it.

File: app/controllers/contactPreference/EmailVerificationController.scala [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+            user.isAgent,
+            None
+          ), routes.EmailVerificationController.updateContactPrefEmail().url
+        )
+        Redirect(controllers.email.routes.EmailChangeSuccessController.show())
+      case Left(ErrorModel(CONFLICT, _)) =>
+        logWarn("[ConfirmWebsiteController][updateWebsite] - There is a contact details update request " +
+          "already in progress. Redirecting user to manage-vat overview page.")
+        Redirect(appConfig.manageVatSubscriptionServicePath)
+          .addingToSession(inFlightContactDetailsChangeKey -> "website")
```
[the remark this answers is not in the document]
Comment (TheDeadlyPianist) [reply app/controllers/contactPreference/EmailVerificationController.scala:112]: Just noticed this. Must have forgot to delete it when I copied it over.

File: app/controllers/contactPreference/EmailVerificationController.scala [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+            Redirect(routes.EmailVerificationController.checkVerificationStatus())
+        }
+        case _ => handleNoEmail
+      }
+  }
+
+  private[controllers] def sendUpdateRequest(email: String)(implicit user: User[_]): Future[Result] = {
+    vatSubscriptionService.updateContactPrefEmail(user.vrn, email).map {
+      case Right(_) =>
+        auditService.extendedAudit(
```
[the remark this answers is not in the document]
Comment (miranda-hawkes) [reply app/controllers/contactPreference/EmailVerificationController.scala:106]: I can still see the audit event in the branch 🤔

File: app/controllers/contactPreference/EmailVerificationController.scala [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+            None
+          ), routes.EmailVerificationController.updateContactPrefEmail().url
+        )
+        Redirect(controllers.email.routes.EmailChangeSuccessController.show())
+      case Left(ErrorModel(CONFLICT, _)) =>
+        logDebug("[ConfirmWebsiteController][updateWebsite] - There is a contact details update request " +
+          "already in progress. Redirecting user to manage-vat overview page.")
+        Redirect(appConfig.btaAccountDetailsUrl)
+      case Left(error) =>
+        logWarn(s"[ConfirmWebsiteController][updateWebsite] - ${error.status}: ${error.message}")
```
Comment (miranda-hawkes) [inline app/controllers/contactPreference/EmailVerificationController.scala:121]: Sorry I've just noticed this
```suggestion
        logDebug("[EmailVerificationController][sendUpdateRequest] - There is a contact details update request " +
          "already in progress. Redirecting user to BTA account details page.")
        Redirect(appConfig.btaAccountDetailsUrl)
      case Left(error) =>
        logWarn(s"[EmailVerificationController][sendUpdateRequest] - ${error.status}: ${error.message}")
```

**Pull request merged**